src/hotspot/share/prims/jvmtiTagMap.hpp

```diff
@@ -27,13 +27,28 @@
 #ifndef SHARE_PRIMS_JVMTITAGMAP_HPP
 #define SHARE_PRIMS_JVMTITAGMAP_HPP
 
+#include "gc/shared/markBitMap.hpp"
 #include "jvmtifiles/jvmti.h"
 #include "memory/allocation.hpp"
 
 class JvmtiEnv;
 class JvmtiTagMapTable;
 class JvmtiTagMapEntryClosure;
 
+// ObjectMarker provides the mark and visited functions
+class ObjectMarker : AllStatic {
+private:
+  static MarkBitMap _mark_bit_map;
+  static MemRegion  _bitmap_region;
+public:
+  static void initialize(MemRegion heap_region);
```

Comment:
So, initialize() runs at VM startup, init() runs when the heap walk starts, if ever?

I'd rename init() to prepare_heapwalk or something similar. initialize vs init is not clear.